packages/components/src/unit-control/utils.ts

```diff
@@ -171,32 +163,35 @@ export function getParsedValue( value, unit, units ) {
 /**
  * Checks if units are defined.
  *
- * @param {any} units Units to check.
- * @return {boolean} Whether units are defined.
+ * @param  units Units to check.
+ * @return Whether units are defined.
  */
-export function hasUnits( units ) {
-	return ! isEmpty( units ) && units !== false;
+export function hasUnits( units: Array< WPUnitControlUnit > | false ): boolean {
```

Comment:
We could also extract `Array< WPUnitControlUnit > | false` type to `types.ts`, to make this pattern more clear. Maybe something like

```ts
type WPUnitControlUnitList = Array< WPUnitControlUnit > | false;
```

This would also help us when potentially working on https://github.com/WordPress/gutenberg/pull/35138#discussion_r717246773